otelcol/collector.go

```diff
@@ -109,14 +110,17 @@ type Collector struct {
 	signalsChannel chan os.Signal
 	// asyncErrorChannel is used to signal a fatal error from any component.
 	asyncErrorChannel chan error
+
+	ol *observer.ObservedLogs
 }
 
 // NewCollector creates and returns a new instance of Collector.
 func NewCollector(set CollectorSettings) (*Collector, error) {
 	var err error
 	configProvider := set.ConfigProvider
 
-	set.ConfigProviderSettings.ResolverSettings.ProviderSettings = confmap.ProviderSettings{Logger: zap.NewNop()}
+	core, ol := observer.New(zap.DebugLevel)
+	set.ConfigProviderSettings.ResolverSettings.ProviderSettings = confmap.ProviderSettings{Logger: zap.New(core)}
 	set.ConfigProviderSettings.ResolverSettings.ConverterSettings = confmap.ConverterSettings{}
```

Comment:
Are we intentionally not passing it to the converter settings?

Reply:
`ConverterSettings` currently [has no fields](https://github.com/open-telemetry/opentelemetry-collector/blob/main/confmap/converter.go#L11) right now. I'd be okay adding it in a separate PR, but adding it here is fine too.

Reply:
My bad, I assumed it did have it. We can do it on a separate PR